packages/leemons-cache/src/types/redis/queries.js

```diff
@@ -1,14 +1,26 @@
 const { isEmpty } = require('lodash');
 
 function generateKey(key, pluginName) {
-  return `${pluginName}.${key}`;
+  return `{plugin.${pluginName}}.${key}`;
 }
 
 function cleanKey(key, pluginName) {
-  return key.replace(new RegExp(`^${pluginName}\\.`), '');
+  return key.replace(new RegExp(`^\\{plugin\\.${pluginName}\\}\\.`), '');
 }
 
-module.exports = function queries(client) {
+module.exports = function queries(client, { isCluster }) {
+  const listKeys = async (pattern) => {
+    if (isCluster) {
+      const allKeys = await Promise.all(
+        client.shards.map(async (shard) => shard.master.client.keys(pattern))
+      );
+
+      return Array.from(new Set(allKeys.flat()));
+    }
+
+    return client.keys(pattern);
```

Comment:
Currently, redis clusters do not support client.keys, so we need to iterate over each cluster and fetch the keys.

> This behaviour can be optimized at the cost of failing while node migrations:
>> I could have predicted the node where the key hash is gonna be and perform on that node the keys operation.